staker-rewards/lib/assets-calculator.js

```diff
@@ -124,14 +124,48 @@ export default class AssetsCalculator {
 
   async calculateKeepStaked(operator) {
     const block = this.interval.endBlock
+    const timestamp = this.interval.end
 
     const operatorContract = this.bondedECDSAKeepFactory.options.address
 
-    const keepStaked = await callWithRetry(
+    let keepStaked = await callWithRetry(
       this.tokenStaking.methods.activeStake(operator, operatorContract),
       block
     )
 
+    // If keep staked (active stake) checked against the factory contract is
+    // zero, there is a possibility the operator undelegated. We need to check
+    // whether it still has a locked active stake in one of the keep contracts.
+    // If so, we count that stake as keep staked.
+    if (keepStaked === "0") {
+      const locks = await callWithRetry(
+        this.tokenStaking.methods.getLocks(operator),
+        block
+      )
+
+      for (let i = 0; i < locks.creators.length; i++) {
+        if (timestamp < locks.expirations[i]) {
```

Comment:
I don't think we should take into account expirations. Even if the lock expired, a deposit can still be active.

Reply:
I think we have no choice and we have to check expiration. If the lock expires, activeStake for that keep will return 0, no matter the underlying keep is active or not

Reply:
This will probably lead to another corner case. We do not need to solve it in this PR but let's work on it right after this one is merged. It does make sense from a contract perspective to do not mark stake as locked if the lock expired but it is still possible that some deposits were not redeemed after 6 months and ETH is bonded. I think we should add a check that if there is a lock and that lock expired, we fallback to `TokenStaking.balanceOf`.

Reply:
Addressed in the follow-up PR: https://github.com/keep-network/keep-ecdsa/pull/790